CondorMonitoring/JobCounter.py

```diff
@@ -589,72 +604,99 @@ def main():
+                    workflow = None
+                    task = None
+                    if schedd_type == 'prodschedd':
+                        workflow = job['WMAgent_SubTaskName'].split('/')[1]
+                        task = job['WMAgent_SubTaskName'].split('/')[-1]
+                        jType = job['CMS_JobType']
+                    elif schedd_type == 'crabschedd':
+                        workflow = re.sub('[:]', '', job['CRAB_UserHN'])
+                        task = job['CRAB_ReqName']
+                        jType = 'Crab3'
+                        if job['TaskType'] == 'ROOT':
+                            # Means it is a ROOT task, which is running only on scheduler.
+                            # and it should not be accounted.
+                            continue
+                    else:
+                        print 'How did you got here?! Are you developing something?'
+                        raise
+                    try:  # it can be an ExprTree
+                        cpus = get_int(job['RequestCpus'])
+                    except:
+                        # Catch any except in case something in the future would change in HTCondor or how ExprTree is evaluated.
+                        # It is not correct to assume it is 1 cpu. Skip this job.
+                        print 'Failed to extract RequestCpus from this job %s' % job
+                        continue
+                    siteToExtract = str(job['DESIRED_Sites']).replace(' ', '').split(",")
+                    while '' in siteToExtract:
+                        siteToExtract.remove('')
+                    if not siteToExtract:
+                        # There are some cases in CRAB, which it makes to have zombie jobs without any DESIRED_Sites.
+                        # See here: https://github.com/dmwm/CRABServer/issues/4933
+                        # Skip it as it will not be able to run anywhere..
+                        print 'No Sites... %s' % job
                         continue
 
-                if status == 2:  # Running
-                    increaseRunning(siteToExtract[0], schedd_name, jType, cpus)
-                    increaseRunningWorkflow(workflow, siteToExtract[0], 1)
-                elif status == 1:  # Pending
-                    pendingCache.append([schedd_name, jType, cpus, siteToExtract])
-                    increasePendingWorkflow(workflow, siteToExtract, 1)
-                else:  # Ignore jobs in another state
-                    continue
+                    if schedd_name in relvalAgents:  # If RelVal job
+                        jType = 'RelVal'
+                    elif task == 'Reco':  # If PromptReco job (Otherwise type is Processing)
+                        jType = 'Reco'
+                    elif jType not in jobTypes:  # If job type is not standard
+                        jType = jobType(jobId, schedd_name, task)
+
+                    siteRunning = job.get(jobKeys[schedd_type]['sitename'], '')
```

Comment:
Also small fixes which are known to be in CRAB or known in HTCondor as partially running (meaning it is in transfer state.
From this elog https://cms-logbook.cern.ch/elog/GlideInWMS/5094
```
Sometimes there are zombie jobs, which do nothing. JobStatus is running, runtime is not increasing over 
time, they show as connected to a pilot from globalpool, but they do not have any MATCH_GLIDEIN_Entry_Name 
or any other attribute, except RemotePool. After some time, sometimes seconds, sometimes minutes, they 
start to have MATCH_GLIDEIN_Entry_Name and their runtime start to increase. From where this delay comes? 
All these kind of jobs are matched to T9_MIT_ZombieNoRunningSite
```
So you cannot assume on one attribute that job is really running there. Both has to be present, so jobStatus=2 and siteRunning is not empty. Sometimes this happens with crab, but as I already have fixed this in cms-gwmsmon for all (crab/prod), they are not easily noticeable.